content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,169 @@
+
+### 3. Notification
+
+During the notification phase, any alteration in a signal's value sets off a chain reaction. 
+The altered signal proactively notifies all its subscribed effects about the modification. 
+Upon receiving this notification, these effects then update or activate as necessary to keep the system's state current and synchronized with the latest data changes.
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity introduces a layer of flexibility and adaptability. 
+Unlike it synchronous counterpart, the updates to the effects are not instantaneous. 
+Instead, they may be delayed or might wait for the completion of an external event or task. 
+This approach is particularly useful in scenarios where immediate updates are not feasible or when the system needs to accommodate external processes before reflecting changes, such as waiting for an API response.
+
+## Efficient data management
+
+### Memo
+
+Memo, short for memoization, is a technique that is used in reactive systems to enhance performance. 
+
+In the context of reactivity, memos become valuable with their ability to selectively recalculate based on changes in specific signals. 
+This means that a computed value or function can intelligently determine when to recalculate based on the signals it depends on. 
+
+#### Caching
+
+This approach involves the caching of results obtained from resource-intensive function calls. 
+Instead of recalculating the output each time a function is invoked, the cached result is returned when the same inputs are encountered again. 
+
+By remembering the last computed value and the conditions (or signal values) under which it was computed, the system can avoid unnecessary recalculations.
+
+#### Dependency tracking
+
+An essential aspect of effective memoization in a reactive system is accurate dependency tracking.  
+This involves keeping track of the signals upon which a computed value relies. 
+This way, the system can identify when the cached value needs to be invalidated and recomputed. 
+
+This fine-grained dependency tracking ensures that the system's reactivity remains reliable and accurate.
+
+### Store
+
+Stores are important as they provide a structured and centralized approach to state management. 
+
+A store in reactive programming is essentially a container for state. 
```

Comment:
Signals update "atomically", it is not possible to listen to only "part" of a signal. Stores, however, allow subscriptions to a specific part of the store, and will only trigger if that part of the store is updated. Therefore, stores are quite common when dealing with complex state.

Aside:
Internally, this is implemented using a tree of nested signals.

Aside:
When setting the store, it is important to only update the part of the store that has changed, as stores do not check whether data has actually changed before triggering. If you need to set a store without knowing exactly what changed, use the `reconcile` helper to diff the data.